codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperFuelWorker.res

```diff
@@ -464,10 +474,28 @@ module Make = (
           }
         | (_, Mint({val, subId}))
         | (_, Burn({val, subId})) =>
-          {
-            "subId": subId,
-            "amount": val,
-          }->Obj.magic
+          (
+            {
+              subId,
+              amount: val,
+            }: Types.fuelSupplyParams
+          )->Obj.magic
+        | (_, TransferOut({amount, assetId, toAddress})) =>
+          (
+            {
+              to: toAddress->Address.unsafeFromString,
+              assetId,
+              amount,
+            }: Types.fuelTransferParams
+          )->Obj.magic
+        | (_, Call({amount, assetId})) =>
+          (
+            {
+              to: contractAddress,
+              assetId,
+              amount,
+            }: Types.fuelTransferParams
+          )->Obj.magic
```

Comment:
How does it work mixing types here? Does the users handler still get the correct type?